rust/src/lib.rs

```
@@ -0,0 +1,343 @@
+
+            // Depending on the mode we're in, we want to handle them differently
+            // In the "starting" mode, we're waiting for the "Starting!" message
+            // from the device.
+
+            // First match just KISS because the rest are all ASCII
+            let state = local_state.load(Ordering::Relaxed);
+
+            if state == MeshTncState::KISS {
+                // Just handle the packet
+            }
+            
+            else {
+                // Convert it to ascii and trim any whitespace
+                let line = String::from_utf8(line)?;
+
+                match state {
+                // This is the origin state where we assume the device has just
+                // booted.  We look for "Starting!" and move into idle.
+                MeshTncState::Starting => {
+                    if line.trim_ascii() == "Starting!" {
+                        // Send the command to set the radio data.
+                        let radio_set = format!(
+                            "set radio {},{},{},{},{}",
+                            self.freq,
+                            self.bw,
+                            self.sf,
+                            self.cr,
+                            self.sb
+                        );
+                        // Send the radio parameters
+                        writer.send(radio_set.as_bytes().to_vec()).await?;
+                        local_state.store(MeshTncState::RadioSet, Ordering::Relaxed);
+                        debug!("Moved into the Radio Set state")
+                    }
+                }
+                MeshTncState::RadioSet => {
+                    // Ensure that the radio parameters were accepted.  We'll
+                    // scan until we get either something that contains "OK"
+                    // or an "Error".  In the case of "Error", the radio didn't
+                    if line.contains("OK") {
+                        // Move onto the "idle" state.
```

you can call `get radio` to ensure that you're in the right mode here

---

Ignore my suggestion for now, it doesn't actually matter what you do here because the underlying code can subtly lie to you.

it doesn't check anything itself, it sets the prefs, and if the prefs pass the constraints, it assumes it worked, but I recently loosened the constraints globally when there needed to be hardware considerations made (for sf5 and 6 support), so i fear that setting the radio settings can actually silently fail right now in specific cases

see here: https://github.com/datapartyjs/MeshTNC/blob/main/src/helpers/CommonCLI.cpp#L327

the get command just reads back the stored prefs, not from the radio itself